packages/experimental/Button/SPEC.md

```diff
@@ -0,0 +1,189 @@
+   */
+  disabled?: boolean;
+
+  /**
+   * A button can format its icon to appear before or after its content.
+   * @default 'before'
+   */
+  iconPosition?: 'before' | 'after';
+
+  // /**
+  //  * A button can show a loading indicator if it is waiting for another action to happen before allowing itself to
+  //  * be interacted with.
+  //  * @default false
+  //  */
+  loading?: boolean;
+
+  /**
+   * A button can be rounded, circular, or square.
+   * @default 'rounded'
+   */
+  shape?: 'rounded' | 'circular' | 'square';
+
+  /**
+   * A button supports different sizes.
+   * @default 'medium'
+   */
+  size?: 'small' | 'medium' | 'large';
+};
+```
+
+### Styling Tokens
+
+TBD once we decide on tokens set.
+
+## Behaviors
+
+### States
+
+The following section describes the different states in which a `Button` can be throughout the course of interaction with it.
+
+#### Enabled state
+
+An enabled `Button` communicates interaction by having styling that invites the user to click/tap on it to trigger an action.
+
+#### Disabled state
+
+A disabled `Button` is non-interactive, disallowing the user to click/tap on it to trigger an action.
+
+#### Hovered state
+
+A hovered `Button` changes styling to communicate that the user has placed a cursor above it.
+
+#### Focused state
+
+A focused `Button` changes styling to communicate that the user has placed keyboard focus on it. This styling is usually the same to the one in the hovered state plus extra styling on the outline to indicate keyboard focus has been placed on the component.
+
+#### Pressed state
+
+A pressed `Button` changes styling to communicate that the user is currently pressing it.
+
+#### Loading state
+
+A loading `Button` renders a `loader` before all the other content to indicate that it is waiting for another action before allowing itself to be interacted with.
+
+### Interaction
+
+#### Keyboard interaction
+
+The following is a set of keys that interact with the `Button` component:
+
+| Key     | Description                   |
+| ------- | ----------------------------- |
+| `Enter` | Executes the `Button` action. |
+| `Space` | Executes the `Button` action. |
+
+#### Cursor interaction
+
+- `mouseenter`: Should immediately change the styling of the `Button` so that it appears to be hovered.
+- `mouseleave`: Should immediately remove the hovered styling of the `Button`.
+- `mouseup`: If triggered while cursor is still inside of the `Button's` boundaries, then it should execute the `Button` and move focus to its target.
+
+#### Touch interaction
+
+The same behavior as above translated for touch events. This means that there is no equivalent for `mouseenter` and `mouseleave`, which makes it so that the hovered state cannot be accessed.
+
+## Accessibility
+
+### Expected behavior
+
+- Should default to adding `role="button"` to the root slot.
+- Should mix in the accessibility props expected for a `button` component.
+- Should be keyboard tabbable and focusable.
```

Comment:
For the future, we could link the `useButton` hook to say exactly what accessibility props are added. Something like that.

Reply:
Added link to useButton hook